### PR TITLE
libtorrent-devel: fix build with gcc for 32-bit

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Use atomic builtins with gcc.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: net/libtorrent-devel/Portfile [hunk 1]
```diff
@@ -49,8 +49,19 @@ post-destroot {
...
 
-# FIXME: 32-bit systems need to switch to C++11 atomics instead of __sync* builtins:
 # https://github.com/rakshasa/libtorrent/issues/163
+if {[string match macports-gcc* ${configure.compiler}]} {
+    patchfiles-append \
+                    0001-Fix-atomics-for-modern-gcc.patch
+
+    # thread_base.cc: error: invalid conversion from 'const char*' to 'char*' [-fpermissive]
+    configure.cxxflags-append \
+                    -fpermissive
```
Comment (pmetzger) [inline net/libtorrent-devel/Portfile:59]: Why do that instead of actually fixing the problem or putting a coercion into the code?

**Comment by reneeotten** [comment]
@barracuda156 it looks like upstream released a new version; did any of this get incorporated there (I don't see a link to an upstream PR/issue in your patch, so who knows if this was sent upstream). It that didn't happen please submit it upstream and see what they way.

File: net/libtorrent-devel/Portfile (hunk 1 above)
Comment (barracuda156) [reply net/libtorrent-devel/Portfile:59]: Cannot check at the moment, but I think this is SDK-related. We cannot control SDK headers.

**Comment by barracuda156** [comment]
> @barracuda156 it looks like upstream released a new version; did any of this get incorporated there (I don't see a link to an upstream PR/issue in your patch, so who knows if this was sent upstream). It that didn't happen please submit it upstream and see what they way.

The problem is still there, from what I can see. I was actually thinking it was reported earlier, but apparently that was indeed missed.
I have opened an issue with upstream now: https://github.com/rakshasa/libtorrent/issues/253

**Comment by barracuda156** [comment]
@reneeotten Yes, this should wait until the upstream hopefully makes a fix.

**Pull request closed without merging**